### PR TITLE
zf: format, update

### PR DESCRIPTION
**Pull request by mmlb**
Format with nixfmt, update 0.9.2 -> 0.10.2

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [x] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [x] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#linking-nixos-module-tests-to-a-package) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) (or backporting [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) and [24.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

**Review by tomodachi94 — CHANGES_REQUESTED** [review]
Hello! The diff looks good, but I encourage you to modify your nixfmt commit (b5bab72882ca6b32d3850595cace78a41e53c71f)  to [use our commit conventions](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#commit-conventions) (prefix it with `zf: `).

Also, is it possible this is a breaking change? From [the changelog](https://github.com/natecraddock/zf/releases/tag/0.10.0):

> The env variable ZF_PROMPT no longer supports terminal escape sequences. It was difficult to support this when rewriting to use libvaxis. Support for this feature may return in the future if there is interest.
> [...]
> Zf no longer does unicode normalization of the input.

If so, could you add a note to `nixos/doc/manual/release-notes/rl-2411.section.md` about it? Just mention that `zf` was updated to 0.10.0 and contained breaking changes (add a link to the changelog for that version).

Thank you!

**Comment by mmlb** [comment]
@tomo thanks, those are good calls will update.

**Review by tomodachi94 — CHANGES_REQUESTED** [review]
Thank you! One little thing and we can get this over the finish line.

File: nixos/doc/manual/release-notes/rl-2505.section.md [hunk 1]
```diff
@@ -69,6 +69,9 @@
   add `vimPlugins.notmuch-vim` to your (Neo)vim configuration if you want the
   vim plugin.
 
+- `zf` no longer does unicode normaliztion of the input and no longer supports terminal escape sequences in the `ZF_PROMPT` env variable.
```
[the remark this answers is not in the document]
Comment (mmlb) [reply nixos/doc/manual/release-notes/rl-2505.section.md]: I did have the link in the changelog but I like your wording better, tweaked it a little bit to have the correct version numbers/link though.

**Comment by mmlb** [comment]
@tomodachi94 ptal, had to rebase on latest master to handle a merge conflict in the docs.

**Review by tomodachi94 — APPROVED** [review]
Lgtm, thank you for your patience ❤️

**Pull request merged**